src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.UnitTests/Services/WhenICreateACsvFile.cs

```diff
@@ -0,0 +1,86 @@
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.ProviderCommitments.Services;
+
+namespace SFA.DAS.ProviderCommitments.UnitTests.Services
+{
+    public class WhenICreateACsvFile
+    {
+        private ICreateCsvService _createCsvService;
+        private List<ApprenticeshipDetails> _apprenticeshipDetails;
+        [SetUp]
+        public void SetUp()
+        {
+            _createCsvService = new CreateCsvService();
+
+            _apprenticeshipDetails = new List<ApprenticeshipDetails>
+            {
+                new ApprenticeshipDetails
+                {
+                    Alerts = null,
+                    ApprenticeName = "Name1",
+                    CourseName = "Course1",
+                    EmployerName = "Employer1",
+                    PlannedEndDateTime = DateTime.UtcNow.AddMonths(2),
+                    PlannedStartDate = DateTime.UtcNow,
+                    Status = "Active",
+                    Uln = "ULN1"
+                },
+                new ApprenticeshipDetails
+                {
+                    Alerts = null,
+                    ApprenticeName = "Name2",
+                    CourseName = "Course2",
+                    EmployerName = "Employer2",
+                    PlannedEndDateTime = DateTime.UtcNow.AddMonths(2),
+                    PlannedStartDate = DateTime.UtcNow,
+                    Status = "Active",
+                    Uln = "ULN2"
+                },
+                new ApprenticeshipDetails
+                {
+                    Alerts = null,
+                    ApprenticeName = "Name3",
+                    CourseName = "Course3",
+                    EmployerName = "Employer3",
+                    PlannedEndDateTime = DateTime.UtcNow.AddMonths(2),
+                    PlannedStartDate = DateTime.UtcNow,
+                    Status = "Active",
+                    Uln = "ULN3"
+                },
+                new ApprenticeshipDetails
+                {
+                    Alerts = null,
+                    ApprenticeName = "Name4",
+                    CourseName = "Course4",
+                    EmployerName = "Employer4",
+                    PlannedEndDateTime = DateTime.UtcNow.AddMonths(2),
+                    PlannedStartDate = DateTime.UtcNow,
+                    Status = "Active",
+                    Uln = "ULN4"
+                }
+            };
+        }
+
+        [Test]
+        public void ThenTheCsvFileContentIsGenerated()
+        {
+            var actual = _createCsvService.GenerateCsvContent(_apprenticeshipDetails);
+
+            Assert.IsNotNull(actual);
+            Assert.IsNotEmpty(actual);
+        }
+
+        [Test]
+        public void AndNothingIsPassedToTheContentGeneratorThenExceptionIsThrown()
+        {
+            List<ApprenticeshipDetails> nullList = null;
+
+            Assert.Throws<WriterException>(() => _createCsvService.GenerateCsvContent(nullList));
+        }
```

Comment:
You could put in a test to confirm that the `actual` returned is actually in CSV format, and a few other things like making sure dates come out correctly for example.